themes/default/source/js/helper.js

```diff
@@ -102,22 +102,28 @@ export const enableScroll = element => {
 };
 
 /**
- * Preload
+ * Preload Images
  * @param  {Array} srcs array of images
+ * @return {Promise}
  */
-export const preloadImages = (srcs, callback) => {
-  let img;
-  let remaining = srcs.length;
-  for (let i = 0; i < srcs.length; i++) {
-    img = new Image();
-    img.onload = () => {
-      --remaining;
-      if (remaining <= 0) {
-        callback();
-      }
-    };
-    img.src = srcs[i];
-  }
+export const preloadImages = (srcs) => {
+  return new Promise((resolve, reject) => {
+    if (!srcs.length) {
+      resolve();
+    }
+    let img;
+    let remaining = srcs.length;
+    for (let i = 0; i < srcs.length; i++) {
+      img = new Image();
+      img.onload = () => {
+        --remaining;
+        if (remaining <= 0) {
+          resolve();
+        }
+      };
+      img.src = srcs[i];
+    }
+  });
```

Comment:
Is there a situation for which we need to `reject` the Promise and `throw` an error? We might loop through the array backwards, like:
```suggestion
    if (!srcs.length) resolve();
    for (let i = srcs.length; i >= 0; --i) {
      const img = new Image();
      img.src = srcs[i];
      img.onload = () => {
        if (i <= 0) resolve();
      };
    }
```